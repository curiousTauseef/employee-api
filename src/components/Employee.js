import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import sanitizeHtml from 'sanitize-html'
import user from '../img/user.jpg'

const MediaObject = styled.li`
    display: flex;
    padding: 20px 0 0 20px;
`

const EmployeeImg = styled.div`
    border-radius: 50%;
    flex-shrink: 0;
    overflow: hidden;
    width: 50px;
    height: 50px;
    object-fit: cover;
`

const EmployeeInfo = styled.div`
    margin-left: 20px;
    border-bottom: 1px solid ${props => props.theme.grey};
    flex-grow: 1;
    padding: 0 20px 20px 0;
`

const Bio = styled.p`
    font-size: 0.875rem;
    color: ${props => props.theme.greyTxt};
`

function addDefaultSrc(e) {
    e.target.src = user
}

const Employee = props => {
    const { employee } = props
    return (
        <MediaObject>
            <EmployeeImg>
                <img
                    src={employee.avatar}
                    alt={employee.name}
                    width="50px"
                    onError={addDefaultSrc}
                />
            </EmployeeImg>
            <EmployeeInfo>
                <h2>{employee.name}</h2>
                <p>
                    {employee.title} at {employee.company}
                </p>
                {Number.isNaN(Number(employee.bio)) && (
                    <Bio>{sanitizeHtml(employee.bio, { allowedTags: [] })}</Bio>
                )}
            </EmployeeInfo>
        </MediaObject>
    )
}

Employee.propTypes = {
    employee: PropTypes.shape({
        uuid: PropTypes.string,
        avatar: PropTypes.string,
        bio: PropTypes.string,
        company: PropTypes.string,
        title: PropTypes.string,
        name: PropTypes.string,
    }).isRequired,
}

export default Employee
