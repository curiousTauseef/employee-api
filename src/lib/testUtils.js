const mockData = [
    {
        uuid: '123',
        company: 'Streich PLC',
        bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
        name: 'Benjamin Shields',
        title: 'Real Estate Sales Agent',
        avatar: 'http://httpstat.us/503',
    },
    {
        uuid: '456',
        company: 'Landgraab Industries',
        bio: '<p>Etiam ut mi sed neque elementum rhoncus.</p>',
        name: 'Gregory Landgraab',
        title: 'CEO',
        avatar: 0,
    },
    {
        uuid: '789',
        company: 'Streich PLC',
        bio: '<pre>Ut eget dui eleifend, pretium dui at, posuere dui.',
        name: 'Bella Goth',
        title: 'Test Subject',
        avatar: 'http://placekitten.com/50/50',
    },
]

module.exports = {
    mockData,
}
