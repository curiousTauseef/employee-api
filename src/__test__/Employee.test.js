import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Employee from '../components/Employee'
import { mockData } from '../lib/testUtils'

describe('<Employee/>', () => {
    it('Should render an employee', () => {
        const wrapper = shallow(<Employee employee={mockData[0]} />)
        expect(wrapper.find('h2').text()).toEqual(mockData[0].name)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })
})
