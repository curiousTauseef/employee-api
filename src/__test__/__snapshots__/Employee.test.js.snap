// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Employee/> Should render an employee 1`] = `
<styled.li>
  <styled.div>
    <img
      alt="Benjamin Shields"
      onError={[Function]}
      src="http://httpstat.us/503"
      width="50px"
    />
  </styled.div>
  <styled.div>
    <h2>
      Benjamin Shields
    </h2>
    <p>
      Real Estate Sales Agent
       at 
      Streich PLC
    </p>
    <styled.p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </styled.p>
  </styled.div>
</styled.li>
`;
